Core/automation/lib/python/core/metadata.py

```diff
@@ -0,0 +1,235 @@
+        log.debug("Deleting metadata namespace '{namespace}' from Item '{item}'".format(
+            item=item_name, namespace=namespace))
+        metadata_registry.remove(MetadataKey(namespace, item_name))
+
+def get_value(item_name, namespace):
+    """
+    Returns the metadata value for the namespace.
+
+    Args:
+        item_name (string): Name of the item.
+        namespace (string): Name of the namespace.
+
+    Raises:
+        ValueError: Item does not exist.
+        ValueError: Namespace name is invalid.
+    """
+    metadata = get_metadata(item_name, namespace)
+    return metadata.value
+
+def set_value(item_name, namespace, value):
+    """
+    Sets the metadata value for the namespace.
+
+    Args:
+        item_name (string): Name of the item.
+        namespace (string): Name of the namespace.
+        value: New value for namespace.
+
+    Raises:
+        ValueError: Item does not exist.
+        ValueError: Namespace name is invalid.
+    """
+    metadata = get_metadata(item_name, namespace)
+    set_metadata(item_name, namespace, value, metadata.configuration)
+
+def get_key_value(item_name, namespace, key):
+    """
+    Fetches the configuration value for the key provided.
+
+    Args:
+        item_name (string): Name of the item.
+        namespace (string): Name of the namespace.
+        key (string): Key to fetch from the metadata configuration.
+
+    Raises:
+        ValueError: Item does not exist.
+        ValueError: Namespace name is invalid.
+    """
+    metadata = get_metadata(item_name, namespace)
+    return metadata.configuration.get(key)
+
+def set_key_value(item_name, namespace, key, value):
+    """
+    Sets a metdata configuration value in the namespace.
+
+    Args:
+        item_name (string): Name of the item.
+        namespace (string): Name of the namespace.
+        key (string): Configuration key to set.
+        value: Value for configuration key.
+
+    Raises:
+        ValueError: Item does not exist.
+        ValueError: Namespace name is invalid.
+    """
+    metadata = get_metadata(item_name, namespace)
+    new_configuration = dict(metadata.configuration).copy()
+    new_configuration.update({key: value})
+    set_metadata(item_name, namespace, metadata.value, new_configuration)
+
+def remove_key_value(item_name, namespace, key):
+    """
+    Removes a configuration key from the metadata namespace.
+
+    Args:
+        item_name (string): Name of the item.
+        namespace (string): Name of the namespace.
+        key (string): Configuration key to remove from the metadata.
+
+    Raises:
+        ValueError: Item does not exist.
+        ValueError: Namespace name is invalid.
+    """
+    metadata = get_metadata(item_name, namespace)
+    new_configuration = dict(metadata.configuration).copy()
+    new_configuration.pop(key, None)
+    set_metadata(item_name, namespace, metadata.value, new_configuration, True)
```

Comment:
value needs to come after configuration in set_metadata

and is there a blank line at the end?

Reply:
Right, we switched value to be kw.

I thought I put a blank line, I'll make sure I'm the edit